src/linear_algebra/basis.lean

```diff
@@ -263,48 +275,64 @@ lemma range_reindex : set.range (b.reindex e) = set.range b :=
 by rw [coe_reindex, range_reindex']
 
 /-- `b.reindex_range` is a basis indexed by `range b`, the basis vectors themselves. -/
-def reindex_range [nontrivial R] : basis (range b) R M :=
-b.reindex (equiv.of_injective b b.injective)
+def reindex_range : basis (range b) R M :=
+if h : nontrivial R then
+  by letI := h; exact b.reindex (equiv.of_injective b (basis.injective b))
+else
+  by letI : subsingleton R := not_nontrivial_iff_subsingleton.mp h; exact basis.map
+  basis.inhabited.default (module.subsingleton_equiv R M (range b)).symm
 
 lemma finsupp.single_apply_left {α β γ : Type*} [has_zero γ]
   {f : α → β} (hf : function.injective f)
   (x z : α) (y : γ) :
   finsupp.single (f x) y (f z) = finsupp.single x y z :=
 by simp [finsupp.single_apply, hf.eq_iff]
 
-lemma reindex_range_self [nontrivial R] (i : ι) (h := set.mem_range_self i) :
+lemma reindex_range_self (i : ι) (h := set.mem_range_self i) :
   b.reindex_range ⟨b i, h⟩ = b i :=
-by rw [reindex_range, reindex_apply, equiv.apply_of_injective_symm b b.injective, subtype.coe_mk]
+begin
+  by_cases htr : nontrivial R,
```

Comment:
You might be able to use the `nontriviality` tactic to automatically do the not-nontrivial case for you.